GraphBase/src/graph/Vertex.java

```diff
@@ -0,0 +1,49 @@
+package graph;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+public class Vertex {
+
+  private int vertexId;
+  private Map<Integer, Edge> adjEdges;
+
+  public Vertex(int newVertexId) {
+    this.vertexId = newVertexId;
+    this.adjEdges = new HashMap<Integer, Edge>();
+  }
+
+  public int getId() {
+    return vertexId;
+  }
+
+  public Map<Integer, Edge> getAdjEdges() {
```

Comment:
I would recommend an api to return all neighbours.
public Collection\<Integer\> getNeighbours() {...}